File: 4200-reverse-letters-then-special-characters-in-a-string/reverse-letters-then-special-characters-in-a-string.java
class Solution {
    public String reverseByType(String s) {
        List<Character> letter=new ArrayList<>();
        List<Character> special=new ArrayList<>();
        char[] ch=s.toCharArray();
        for(char c:ch){
            if(Character.isLetter(c)){
                letter.add(c);
            }
            else{
                special.add(c);
            }
        }
        Collections.reverse(letter);
        Collections.reverse(special);
        int j=0;
        int k=0;
        for(int i=0;i<ch.length;i++){
            if(Character.isLetter(ch[i])){
                ch[i]=letter.get(j);
                j++;
            }
            else{
                ch[i]=special.get(k);
                k++;
            }
        }
        return new String(ch);
    }
}

File: 4200-reverse-letters-then-special-characters-in-a-string/reverse-letters-then-special-characters-in-a-string.cpp
class Solution {
public:
    string reverseByType(string s) {
        vector<char> letter;
        vector<char> special;
        for(char ch:s){
            if(isalpha(ch)){
                letter.push_back(ch);
            }
            else{
                special.push_back(ch);
            }
        }
        reverse(letter.begin(),letter.end());
        reverse(special.begin(),special.end());
        int j=0;
        int k=0;
        for(int i=0;i<s.length();i++){
            if(isalpha(s[i])){
                s[i]=letter[j];
                j++;
            }
            else{
                s[i]=special[k];
                k++;
            }
        }
        return s;
    }
};